bigtable/tests/mutations_integration_test.cc

```diff
@@ -0,0 +1,555 @@
+   * Delete records from a specific row_key and specific column_family
+   */
+  void DeleteFromFamily(bigtable::Table& table, std::string row_key,
+                        std::string column_family) {
+    bigtable::SingleRowMutation mutation(row_key);
+    mutation.emplace_back(bigtable::DeleteFromFamily(column_family));
+
+    bigtable::BulkMutation bulk;
+    bulk.emplace_back(std::move(mutation));
+    table.BulkApply(std::move(bulk));
+  }
+
+  /**
+   * Delete records from a specific row key
+   */
+  void DeleteFromRow(bigtable::Table& table, std::string row_key) {
+    bigtable::SingleRowMutation mutation(row_key);
+    mutation.emplace_back(bigtable::DeleteFromRow());
+
+    bigtable::BulkMutation bulk;
+    bulk.emplace_back(std::move(mutation));
+    table.BulkApply(std::move(bulk));
+  }
+};
+
+}  // namespace anonymus
+
+/**
+ *  Check if the values inserted by SetCell are correctly inserted into
+ *  Cloud Bigtable
+ */
+TEST_F(MutationIntegrationTest, SetCellTest) {
+  std::string const table_name = "table-setcell";
+
+  auto table = CreateTable(table_name, table_config);
+  // Create a vector of cells which will be inserted into bigtable
+  std::string const row_key = "SetCellRowKey";
+  std::vector<bigtable::Cell> created_cells{
+      {row_key, column_family1, "column_id1", 0, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id1", 1000, "v-c-0-1", {}},
+      {row_key, column_family1, "column_id1", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 0, "v-c0-0-0", {}},
+      {row_key, column_family2, "column_id3", 1000, "v-c1-0-1", {}},
+      {row_key, column_family3, "column_id1", 2000, "v-c1-0-2", {}},
+  };
+
+  CreateCells(*table, created_cells);
+
+  auto actual_cells = ReadRows(*table, bigtable::Filter::PassAllFilter());
+
+  DeleteTable(table_name);
+
+  CheckEqualUnordered(created_cells, actual_cells);
+}
+
+/**
+ *  Verify that the values inserted by SetCell with server-side timestamp are
+ *  correctly inserted into Cloud Bigtable.
+ */
+TEST_F(MutationIntegrationTest, SetCellIgnoreTimestampTest) {
+  std::string const table_name = "table-setcell-ignore-timestamp";
+
+  auto table = CreateTable(table_name, table_config);
+
+  // Create a vector of cell which will be inserted into bigtable
+  std::string const row_key = "SetCellRowKey";
+  std::vector<bigtable::Cell> created_cells{
+      {row_key, column_family1, "column_id1", 0, "v-c-0-0", {}},
+      {row_key, column_family1, "column_id2", 1000, "v-c-0-1", {}},
+      {row_key, column_family1, "column_id3", 2000, "v-c-0-2", {}},
+      {row_key, column_family2, "column_id2", 0, "v-c0-0-0", {}},
+      {row_key, column_family2, "column_id3", 1000, "v-c1-0-1", {}},
+      {row_key, column_family3, "column_id1", 2000, "v-c1-0-2", {}},
+  };
+
+  // This microseconds are 1 milliseconds mismatch with the server timestamp.
+  // So during comparison this timestamp is fixed to 1000 and actual and
+  // expected are matched
+  auto micro_secs =
```

Comment:
I still do not understand why you set a carefully computed `micro_secs` value if you are then going to ignore it.